python_modules/dagster/dagster/_core/host_representation/external.py

```diff
@@ -144,34 +117,18 @@ def get_external_partition_sets(self):
             for external_partition_set_data in self.external_repository_data.external_partition_set_datas
         ]
 
-    def get_full_external_pipeline(self, pipeline_name: str) -> "ExternalPipeline":
-        check.str_param(pipeline_name, "pipeline_name")
-        return ExternalPipeline(
-            self.external_repository_data.get_external_pipeline_data(pipeline_name),
-            repository_handle=self.handle,
-            pipeline_index=self.get_pipeline_index(pipeline_name),
-        )
-
-    def get_all_external_pipelines(self):
-        return [self.get_full_external_pipeline(pn) for pn in self._pipeline_index_map]
-
     def has_external_job(self, job_name):
-        return job_name in self._job_index_map
+        return job_name in self._job_data_map
 
-    def get_external_job(self, job_name) -> "ExternalPipeline":
+    def get_full_external_job(self, job_name: str) -> "ExternalPipeline":
         check.str_param(job_name, "job_name")
-
-        if not self.has_external_job(job_name):
-            check.failed(f"Could not find job data for {job_name}")
-
         return ExternalPipeline(
-            self.external_repository_data.get_external_pipeline_data(job_name),
+            self._job_data_map[job_name],
```

Comment:
do we still need this check to ensure that the `job_name` references an existing job?